synapse/storage/databases/main/appservice.py

```diff
@@ -173,7 +173,7 @@ async def get_appservice_state(
             desc="get_appservice_state",
         )
         if result:
-            return result.get("state")
+            return ApplicationServiceState(result.get("state"))
```

Comment:
Just a small note to say that this changed the behaviour of the function. Previously this function returned `None` if `state` was `null` in the DB, but with this change the function errors instead.

This is fixed in https://github.com/matrix-org/synapse/pull/12680#discussion_r874515829